.github/workflows/ci.yml

```diff
@@ -288,8 +334,8 @@ jobs:
       fail-fast: true
       matrix:
         build:
-          - { os: macos-12,    xcode: 14.2,   deployment: 12.5 } # LLVM14, x86_64
-          - { os: macos-14,    xcode: 15.2,   deployment: 14.0 } # LLVM16, arm64
+          - { os: macos-13,    xcode: 15.2,   deployment: 13.5 } # LLVM16, x86_64
+          - { os: macos-14,    xcode: 15.2,   deployment: 13.5 } # LLVM16, arm64
```

Comment:
Deployment 13.5 for macos-14 will result in lots of linker warnings. We should keep deployment 14 here.